src/parser/offset.rs

```diff
@@ -0,0 +1,398 @@
+        let ret = *self;
+        match self {
+            ByteOffset::Bytes(n_bytes) => {
+                *n_bytes += delta;
+            }
+            ByteOffset::Bits {
+                bits_advanced: n_bits,
+                ..
+            } => {
+                *n_bits += delta;
+            }
+        }
+        ret
+    }
+
+    pub(crate) fn enter_bits_mode(&mut self) -> Result<(), ParseError> {
+        if let ByteOffset::Bytes(n_bytes) = *self {
+            *self = ByteOffset::Bits {
+                starting_byte: n_bytes,
+                bits_advanced: 0,
+            };
+            Ok(())
+        } else {
+            Err(ParseError::InternalError(StateError::BinaryModeError))
+        }
+    }
+
+    pub(crate) fn escape_bits_mode(&mut self) -> Result<usize, ParseError> {
+        if let ByteOffset::Bits {
+            starting_byte,
+            bits_advanced,
+        } = *self
+        {
+            let delta_major = bits_advanced / 8;
+            let delta_minor = bits_advanced % 8;
+            if delta_minor != 0 {
+                *self = ByteOffset::Bytes(starting_byte + delta_major + 1);
+            } else {
+                *self = ByteOffset::Bytes(starting_byte + delta_major);
+            }
+            Ok(bits_advanced)
+        } else {
+            Err(ParseError::InternalError(StateError::BinaryModeError))
+        }
+    }
+
+    pub(crate) fn abs_bit_offset(&self) -> usize {
+        match self {
+            &ByteOffset::Bytes(n) => n * 8,
+            &ByteOffset::Bits {
+                starting_byte,
+                bits_advanced,
+            } => starting_byte * 8 + bits_advanced,
+        }
+    }
+
+    pub(crate) fn bits_advanced(&self) -> Option<usize> {
+        match self {
+            ByteOffset::Bytes(_n) => None,
+            &ByteOffset::Bits { bits_advanced, .. } => Some(bits_advanced),
+        }
+    }
+
+    pub(crate) fn as_bytes(&self) -> (usize, Option<usize>) {
+        match self {
+            &ByteOffset::Bytes(n) => (n, None),
+            &ByteOffset::Bits {
+                starting_byte,
+                bits_advanced,
+            } => {
+                let delta_major = bits_advanced / 8;
+                let delta_minor = bits_advanced % 8;
+                (starting_byte + delta_major, Some(delta_minor))
+            }
+        }
+    }
+}
+
+impl PartialOrd for ByteOffset {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        let partial = self.abs_bit_offset().cmp(&other.abs_bit_offset());
+        match partial {
+            // yield None instead of Some(Equal) if same bit-level offset but different modes
+            Ordering::Equal if self.is_bit_mode() ^ other.is_bit_mode() => None,
+            _ => Some(partial),
+        }
+    }
+}
+
+pub(crate) struct BufferOffset {
+    current_offset: ByteOffset, // the 'true' value of the offset, reinterpreted according to the other fields
+    view_stack: ViewStack,
```

Comment:
Would be good to know what this means, like reinterpreted according to view_stack?

Reply:
This may be an artifact of an earlier version. I frankly don't know *exactly* what it is referring to, and it is probably stale.